src/luascript.cpp

```diff
@@ -12608,6 +13360,383 @@ int LuaScriptInterface::luaPartySetSharedExperience(lua_State* L)
 	return 1;
 }
 
+// Loot
+
+int LuaScriptInterface::luaCreateLoot(lua_State* L)
+{
+	// Loot() will create a new loot item
+	Loot* loot = new Loot();
```

Comment:
The lua GC is not fairy dust, it will NOT and in fact CANNOT cleanup memory that is used by the server (unless it's specifically setup to do so, via the __gc() metamethod). The reason that monster reloads don't affect memory usage in this case is the fact that you're not reloading lua-based monsters.